backpack/utils/conv_transpose.py

```diff
@@ -1,6 +1,44 @@
 import torch
 from torch.nn.functional import conv_transpose1d, conv_transpose2d, conv_transpose3d
 
+from backpack.utils.ein import eingroup
+
+
+def get_convtranspose1d_weight_gradient_factors(input, grad_out, module):
+    N, C_in = input.shape[0], input.shape[1]
+    kernel_size = module.kernel_size
+    kernel_size_numel = int(torch.prod(torch.Tensor(kernel_size)))
+
+    X = (torch.flatten(unfold_by_conv_transpose(input, module))).reshape(
+        N, C_in * kernel_size_numel, -1
+    )
+    dE_dY = eingroup("n,c,l->n,c,l", grad_out)
+    return X, dE_dY
+
+
+def get_convtranspose2d_weight_gradient_factors(input, grad_out, module):
+    N, C_in = input.shape[0], input.shape[1]
+    kernel_size = module.kernel_size
+    kernel_size_numel = int(torch.prod(torch.Tensor(kernel_size)))
+
+    X = (torch.flatten(unfold_by_conv_transpose(input, module))).reshape(
+        N, C_in * kernel_size_numel, -1
+    )
+    dE_dY = eingroup("n,c,h,w->n,c,hw", grad_out)
+    return X, dE_dY
+
+
+def get_convtranspose3d_weight_gradient_factors(input, grad_out, module):
+    N, C_in = input.shape[0], input.shape[1]
+    kernel_size = module.kernel_size
+    kernel_size_numel = int(torch.prod(torch.Tensor(kernel_size)))
+
+    X = (torch.flatten(unfold_by_conv_transpose(input, module))).reshape(
```

Comment:
The `torch.flatten(...)` should not be required. Please check and remove it if possible